### ISSUE TITLE
High resource usage on nomad servers when running replicator

### ISSUE THREAD

**Issue by burdandrei** [elsevier-core-engineering/replicator#223]
**Description**
Running replicator, watching 50 job groups overloading nomad server:
![image](https://user-images.githubusercontent.com/2526486/31081364-5334c1fa-a794-11e7-8499-a0ad24dc17f7.png)

Running all the checks in go routines makes to many calls to servers

**Steps to reproduce the issue:**
1. add meta to 50 jobs
2. run replicator
3.

**Describe the results you expected:**
not to kill it =)
**Describe the results you received:**
High cpu and memory usage

**Additional information you deem important (e.g. issue happens only occasionally):**

**Replicator version:**
v1.0.3
**Output of `consul version`:**
```
Consul v0.9.3
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
```

**Output of `nomad version`:**

```
Nomad v0.6.3
```

**Additional environment details:**

**Debug log outputs from replicator:**

**Comment by ericwestfall** [comment]
@burdandrei to be clear, in the telemetry data, 10.101.5.11 represents a Nomad Server node or the Nomad Client running Replicator?

**Comment by burdandrei** [comment]
@ericwestfall  it's a nomad server

**Comment by ericwestfall** [comment]
This is being addressed in #224 and #221 - full details can be found in each issue.

**Comment by burdandrei** [comment]
:100:
